library_information_validator.py

```diff
@@ -0,0 +1,495 @@
+            else:
+                index_only = re.split('\(|\)', self.access_sample_info_sheet[sindex].value)[1]
+
+                # checks that dual indexes have the same length in both components
+                split_index = re.split('[-_]',index_only)
+                if(len(split_index) > 1):
+                    if(len(split_index[0]) != len(split_index[1])):
+                        logger.warning("Length of the two components in dual index {} and {} is different.".format(split_index[0], split_index[1]))
+                        warning_component_length += 1
+
+                # merges dual indexes to one sequence
+                index_only = re.sub('[-_]', '', index_only)
+                sindex_list.append(index_only)
+
+                # generates error if both custom and NGI standard index are selected for the same sample
+                if (self.access_sample_info_sheet["{col}{row_nr}".format(col=LibrarySheet.CINDEX_COL, row_nr=sindex.split(LibrarySheet.SINDEX_COL)[1])].value is not None):
+                    logger.error('Custom and Standard Index selected for the sample in fields {}{} and {}{}. Please clarify which of the two indexes was used.'\
+                    .format(LibrarySheet.SINDEX_COL, sindex.split(LibrarySheet.SINDEX_COL)[1], LibrarySheet.CINDEX_COL, sindex.split(LibrarySheet.SINDEX_COL)[1]))
+                    quit()
+
+        # retrieves index sequences for custom indexes
+        cindex_list = []
+        cindex_absent = []
+        for cindex in cell_id_custom_index:
+            cindex_value = self.access_sample_info_sheet[cindex].value
+            if (cindex_value is None):
+                cindex_absent.append(cindex)
+            else:
+                # checks that dual indexes have the same length in both components
+                split_cindex = re.split('[-_]',cindex_value)
+                if(len(split_cindex) > 1):
+                    if(len(split_cindex[0]) != len(split_cindex[1])):
+                        logger.warning("Length of the two components in dual index {} and {} is different.".format(split_cindex[0], split_cindex[1]))
+                        warning_component_length += 1
+
+                # merges dual indexes to one sequence
+                index_only = re.sub('[-]','', cindex_value)
+                cindex_list.append(index_only)
+
+        # identifies missing index specification for individual samples
+        # and detects wether NGI standard indexes or custom indexes are chosen for a given pool
+        error_missing_index = 0
+        warning_mixed_indexes = 0
+        sindex_chosen = True
+        if(len(cindex_absent) == len(pool_rows)):
+            sel_index = sindex_list
+            if(len(sindex_list) != len(pool_rows)):
+                for absent_index in sindex_absent:
+                    logger.error("missing index in row {}".format(re.sub("\D","",absent_index)))
+                    quit()
```

Comment:
It's bad Python style to use `quit()` like this. Better to raise an exception (pick one from [here](https://airbrake.io/blog/python-exception-handling/class-hierarchy)), eg. : `raise AssertionError`

If you don't want the messy stack trace, you can catch this exception in the top-level main function and exit there instead. This makes your code more reusable.

If you want to exit cleanly, you should use `sys.exit(1)` instead of `quit()`. `sys.exit` is the proper way to kill the program, and setting `1` as the return code signals that something went wrong (`0` as an exit code means that everything worked).